### PR TITLE
android-tools: install missing tools

### PR DESCRIPTION
**Pull request by musfay**
###### Motivation for this change
Added tools required for tinker with verified boot

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/android-tools/default.nix [hunk 1]
```diff
@@ -23,8 +27,13 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  postPatch = ''
+    sed -i -E "0,/import api_pb2/ s//from google.protobuf import api_pb2/" vendor/avb/aftltool.py
```
Comment (primeos) [inline pkgs/tools/misc/android-tools/default.nix:31]: Could we switch to `substituteInPlace` here? That way it emits a warning if this doesn't match anymore.

File: pkgs/tools/misc/android-tools/default.nix [hunk 2]
```diff
@@ -33,6 +42,12 @@ stdenv.mkDerivation rec {
     export GOCACHE=$TMPDIR/go-cache
   '';
 
+  postInstall = ''
+    install -Dm755 ../vendor/avb/aftltool.py -t $out/bin
+    install -Dm755 ../vendor/avb/avbtool.py -t $out/bin
+    install -Dm755 ../vendor/mkbootimg/mkbootimg.py $out/bin/mkbootimg
```
Comment (primeos) [inline pkgs/tools/misc/android-tools/default.nix:48]: Is there any reason to only remove the `.py` suffix from `mkbootimg` (I've never used those tools).

**Review by primeos — CHANGES_REQUESTED** [review]
Not sure if this is a good idea tbh (we're deviating from upstream here) but I'm ok with giving it a go. One nit and one question though.

File: pkgs/tools/misc/android-tools/default.nix (hunk 1 above)
Comment (musfay) [reply pkgs/tools/misc/android-tools/default.nix:31]: I used `substituteInPlace` first but there is exact two `import api_pb2` and we only need to fix first one. Check [this line](https://android.googlesource.com/platform/external/avb/+/4749d5547d9f5f54b0407d82b5f52ea85c8f0d9c/aftltool.py#1497). `substituteInPlace` replaces both of them.

File: pkgs/tools/misc/android-tools/default.nix (hunk 2 above)
Comment (musfay) [reply pkgs/tools/misc/android-tools/default.nix:48]: `aftltool` imports `avbtool` so we need to keep .py suffix. `aftltool` is not a common tool I don't know we should include it or not.

Comment (musfay) [reply pkgs/tools/misc/android-tools/default.nix:48]: There is a [patch](https://build.opensuse.org/package/view_file/hardware/android-tools/fix-install-python-tools.patch?expand=1) that applied on openSUSE package which also installs these tools and keeps `.py` suffix but not sure OBS is reproducible.

File: pkgs/tools/misc/android-tools/default.nix (hunk 1 above)
Comment (primeos) [reply pkgs/tools/misc/android-tools/default.nix:31]: Ok, that also explains the sed expression then, thanks! Using e.g. `s/^import api_pb2$/from google.protobuf import api_pb2/` should also work then but it's fine either way.

File: pkgs/tools/misc/android-tools/default.nix (hunk 2 above)
Comment (primeos) [reply pkgs/tools/misc/android-tools/default.nix:48]: Ok, should be fine then, thanks!

**Review by primeos — APPROVED** [review]

**Comment by primeos** [comment]
Two upstream links related to this (just to document it here as well):
- https://github.com/nmeum/android-tools/pull/37
  - https://github.com/nmeum/android-tools/pull/10#issuecomment-595775875

Not much information but it looks like they might be open to it.

**Pull request merged**